addons/movments_accounts_balances/controllers/controllers.py

```diff
@@ -79,3 +79,32 @@ def general_ledger_report(self, account_id, start_date, end_date):
             'ledger_data': ledger_data,
         }
 
+    @api.model
+    def create_bill(self, partner_id, invoice_date, due_date, reference, line_data, global_narration):
+        # Retrieve the currency ID for USD
+        currency_usd_id = self.env.ref('base.USD').id
+
+        Bill = self.env['account.move']
+        invoice_lines = []
+        for line in line_data:
+            invoice_line_vals = {
+                'name': line['description'],
+                'quantity': 1,
+                'price_unit': line['amount'],
+                'account_id': line['account_id'],
+                'analytic_account_id': line.get('analytic_account_id', False),  # Add analytic account if provided
+            }
+            invoice_lines.append((0, 0, invoice_line_vals))
+
+        bill = Bill.create({
+            'move_type': 'in_invoice',  # Vendor Bill
+            'partner_id': partner_id,
+            'invoice_date': invoice_date,
+            'invoice_date_due': due_date,
+            'invoice_line_ids': invoice_lines,
```

Comment:
'invoice_line_ids': invoice_lines,This concept is incorrect and cannot be implemented in such a straightforward manner. The only way is to create the 'account.move.line' based on the 'account.move' ID.